### DISCUSSION TITLE
How to filter products by facet values?

### DISCUSSION THREAD

**Discussion by alphonso06** [vendurehq/vendure#2354] [Q&A] [answered]
Hello,

I'm trying to filter all products by facet values using the `products()` query.

Here is my query:
```gql
query Products($take: Int!, $skip: Int!) {
  products(options:{
    take: $take,
    skip: $skip
  }) {
    totalItems
    items {
      slug
      featuredAsset {
        source
      }
      facetValues {
        name
      }
      name
      variants {
        price
      }
    }
  }
}
```

There doesn't seem to be a way using the `ProductListParameter`. I tried using the `search()` query but its results don't yield the facet value names, I need those to be displayed on the front-end.

I might be missing a thing or two here but any help will be greatly appreciated. 😄

**Comment by michaelbromley** [comment, marked as answer]
Hi,

Your observations are correct - the `products` query does not include the ability to filter by facet value, only the `search` query has that functionality out of the box.

Although the `search` query does not include the facet value names, you can get access to them via the `facetValues` field on the search response type. Here's some docs that show this: https://beta-docs.vendure.io/guides/storefront/listing-products/#faceted-search

Here's a live demo of a faceted search: https://stackblitz.com/edit/vendure-docs-faceted-search
